File: examples/js/agents/basic-agent.ts
/**
 * Basic Agent Example
 * Demonstrates simple agent usage with OpenAI
 */

import { Agent } from 'praisonai';

async function main() {
  const agent = new Agent({
    name: 'Assistant',
    instructions: 'You are a helpful assistant. Be concise and informative.',
    llm: 'openai/gpt-4o-mini'
  });

  const response = await agent.chat('What is TypeScript?');
  console.log('Response:', response.text);
}

main().catch(console.error);

File: src/praisonai-rust/examples/basic_agent.rs
//! Basic agent example
//!
//! Run with: cargo run --example basic_agent

use praisonai::Agent;

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    // Create a simple agent
    let agent = Agent::new()
        .name("assistant")
        .instructions("You are a helpful AI assistant. Be concise and friendly.")
        .model("gpt-4o-mini")
        .build()?;
    
    println!("Agent created: {}", agent.name());
    println!("Model: {}", agent.model());
    
    let response = agent.chat("What is 2 + 2?").await?;
    println!("Response: {}", response);
    
    Ok(())
}
